tests/kafkatest/services/kafka/kafka.py

```diff
@@ -227,10 +228,9 @@ def start_node(self, node):
 
         cmd = self.start_cmd(node)
         self.logger.debug("Attempting to start KafkaService on %s with command: %s" % (str(node.account), cmd))
-        with node.account.monitor_log(KafkaService.STDOUT_STDERR_CAPTURE) as monitor:
-            node.account.ssh(cmd)
-            # Kafka 1.0.0 and higher don't have a space between "Kafka" and "Server"
-            monitor.wait_until("Kafka\s*Server.*started", timeout_sec=30, backoff_sec=.25, err_msg="Kafka server didn't finish startup")
+        node.account.ssh(cmd)
+        wait_until(lambda: listening(self.logger, node, self.jmx_port), timeout_sec=30, backoff_sec=.25,
```

Comment:
I think you'd want to use actual listener ports, not the JMX one. The JMX one is presumably opened very early when the process starts, but we want to make sure the Kafka service is actually up, running, and ready to serve traffic. That's why the previous version was checking for a message that happens much later in the startup process.

Reply:
@ewencp Many thanks for pointing this out. I changed kafka.py to use listener port instead of the JMX one.